File: BBIS.Application/DTOs/Module/ModuleDto.cs
namespace BBIS.Application.DTOs.Module
{
    public class ModuleDto
    {
        public Guid ModuleId { get; set; }
        public string Menu { get; set; }
        public string Icon { get; set; }
        public string Link { get; set; }
        public int OrderNo { get; set; }
        public bool IsParentMenu { get; set; }
        public Guid? ParentModuleId { get; set; }

        public List<ModuleDto> SubMenuItems { get; set; }
    }
}

File: BBIS.Api/ClientApp/src/models/UserAccounts/ModuleDto.ts
export interface ModuleDto {
    Icon: string;
    IsParentMenu: boolean;
    Link: string;
    Menu: string;
    ModuleId: Guid;
    OrderNo: number;
    SubMenuItems: Array<ModuleDto>;
    Toggle: boolean;
    ToggleChild: boolean;
    ParentModuleId: Guid | null;
}